equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,248 @@
+package equinix
+
+import (
+	"context"
+	"fmt"
+	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
+	"github.com/equinix/terraform-provider-equinix/internal/config"
+	equinix_schema "github.com/equinix/terraform-provider-equinix/internal/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"strings"
+	"time"
+)
+
+func resourceNetwork() *schema.Resource {
+	return &schema.Resource{
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(6 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(6 * time.Minute),
+			Read:   schema.DefaultTimeout(6 * time.Minute),
+		},
+		ReadContext:   resourceNetworkRead,
+		CreateContext: resourceNetworkCreate,
+		UpdateContext: resourceNetworkUpdate,
+		DeleteContext: resourceNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: createNetworkResourceSchema(),
+
+		Description: "Fabric V4 API compatible resource allows creation and management of Equinix Fabric Network\n\n~> **Note** Equinix Fabric v4 resources and datasources are currently in Beta. The interfaces related to `equinix_fabric_` resources and datasources may change ahead of general availability. Please, do not hesitate to report any problems that you experience by opening a new [issue](https://github.com/equinix/terraform-provider-equinix/issues/new?template=bug.md)",
+	}
+}
+
+func resourceNetworkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	schemaNotifications := d.Get("notifications").([]interface{})
+	notifications := notificationToFabric(schemaNotifications)
+	schemaLocation := d.Get("location").(*schema.Set).List()
+	location := locationToFabric(schemaLocation)
+	project := v4.Project{}
+	schemaProject := d.Get("project").(*schema.Set).List()
+	if len(schemaProject) != 0 {
+		project = projectToFabric(schemaProject)
+	}
+	netType := v4.NetworkType(d.Get("type").(string))
+	netScope := v4.NetworkScope(d.Get("scope").(string))
+
+	createRequest := v4.NetworkPostRequest{
+		Name:          d.Get("name").(string),
+		Type_:         &netType,
+		Scope:         &netScope,
+		Location:      &location,
+		Notifications: notifications,
+		Project:       &project,
+	}
+
+	fabricNetwork, _, err := client.NetworksApi.CreateNetwork(ctx, createRequest)
+	if err != nil {
+		return diag.FromErr(err)
```

Comment:
All error responses from Fabric Go SDK need to be updated to follow our error output pattern. Use this for reference: https://github.com/equinix/terraform-provider-equinix/pull/449

Reply:
Ok, got it. I see that the main branch was updated with this code just a couple of hours ago after my PR was out. Let me check if I can rebase on latest changes.

Reply:
All good! There's no way you could have known. Just wanted to make sure that now we have it we keep up with it. :)